src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -328,6 +330,12 @@ private void MoveNext(ref int index)
             index = (tmp == _array.Length) ? 0 : tmp;
         }
 
+        private void ThrowForEmptyQueue()
```

Comment:
Dequeue can call this as well, right?  If nothing else it'd avoid duplicate usage of InvalidOperation_EmptyQueue.  Not enough of a reason to separate it out into its own function, but as long as it is being separated out, might as well do so.